gsonpath-compiler/src/main/java/gsonpath/model/GsonObjectFactory.kt

```diff
@@ -171,23 +140,123 @@ class GsonObjectFactory(private val subTypeMetadataFactory: SubTypeMetadataFacto
     private fun addStandardType(
             gsonPathObject: GsonObject,
             fieldInfo: FieldInfo,
-            jsonFieldPath: String,
+            jsonFieldPath: FieldPath.Standard,
             fieldInfoIndex: Int,
             isRequired: Boolean,
             gsonSubTypeMetadata: SubTypeMetadata?) {
 
-        if (!gsonPathObject.containsKey(jsonFieldPath)) {
-            gsonPathObject.addField(jsonFieldPath,
-                    GsonField(fieldInfoIndex, fieldInfo, jsonFieldPath, isRequired, gsonSubTypeMetadata))
+        val path = jsonFieldPath.path
+        if (!gsonPathObject.containsKey(path)) {
+            gsonPathObject.addField(path, GsonField(fieldInfoIndex, fieldInfo, path, isRequired, gsonSubTypeMetadata))
 
         } else {
-            throwDuplicateFieldException(fieldInfo.element, jsonFieldPath)
+            throwDuplicateFieldException(fieldInfo.element, path)
         }
     }
 
+    /**
+     * Applies the gson field naming policy using the given field name.
+     * @param fieldNamingPolicy the field naming policy to apply
+     * *
+     * @param fieldName         the name being altered.
+     * *
+     * @return the altered name.
+     */
+    @Throws(ProcessingException::class)
+    private fun applyFieldNamingPolicy(fieldNamingPolicy: FieldNamingPolicy, fieldName: String): String {
+        //
+        // Unfortunately the field naming policy uses a Field parameter to translate name.
+        // As a result, for now it was decided to create a fake field class which supplies the correct name,
+        // as opposed to copying the logic from GSON and potentially breaking compatibility if they add another enum.
+        //
+        val fieldConstructor = Field::class.java.declaredConstructors[0]
+        fieldConstructor.isAccessible = true
+        val fakeField: Field
+        try {
+            fakeField = fieldConstructor.newInstance(null, fieldName, null, -1, -1, null, null) as Field
+
+        } catch (e: InstantiationException) {
+            throw ProcessingException("Error while creating 'fake' field for naming policy.")
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Error while creating 'fake' field for naming policy." 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=kotlin%3AS1192)